emily/handler/src/database/entries/mod.rs

```diff
@@ -1,8 +1,435 @@
+        let key_item: serde_dynamo::Item = serde_dynamo::to_item(key)?;
+        // Query the database.
+        let get_item_output = dynamodb_client
+            .get_item()
+            .table_name(Self::table_name(settings))
+            .set_key(Some(key_item.into()))
+            .send()
+            .await?;
+        // Get DynamoDB item.
+        let item = get_item_output.item.ok_or(Error::NotFound)?;
+        // Convert item into entry.
+        let entry = serde_dynamo::from_item(item)?;
+        // Return.
+        Ok(entry)
+    }
+
+    /// Generic table query for all attributes with a given primary key.
+    async fn query_with_partition_key(
+        dynamodb_client: &aws_sdk_dynamodb::Client,
+        settings: &Settings,
+        partition_key: &<<Self::Entry as EntryTrait>::Key as KeyTrait>::PartitionKey,
+        maybe_next_token: Option<String>,
+        maybe_page_size: Option<i32>,
+    ) -> Result<(Vec<Self::Entry>, Option<String>), Error> {
+        // Convert inputs into the types needed for querying.
+        let exclusive_start_key =
+            maybe_exclusive_start_key_from_next_token::<Self::SearchToken>(maybe_next_token)?;
+        // Query the database.
+        let query_output = dynamodb_client
+            .query()
+            .table_name(Self::table_name(settings))
+            .set_index_name(Self::INDEX_NAME_IF_GSI.map(|s| s.to_string()))
+            .set_exclusive_start_key(exclusive_start_key)
+            .set_limit(maybe_page_size)
+            .key_condition_expression("#pk = :v")
+            .expression_attribute_names(
+                "#pk",
+                <<Self::Entry as EntryTrait>::Key as KeyTrait>::PARTITION_KEY_NAME,
+            )
+            .expression_attribute_values(":v", serde_dynamo::to_attribute_value(partition_key)?)
+            .send()
+            .await?;
+        // Convert data into output format.
+        let entries: Vec<Self::Entry> =
+            serde_dynamo::from_items(query_output.items.unwrap_or_default())?;
+        let next_token = maybe_next_token_from_last_evaluated_key::<Self::SearchToken>(
+            query_output.last_evaluated_key,
+        )?;
+        // Return.
+        Ok((entries, next_token))
+    }
+
+    /// Generic put table entry.
+    async fn put_entry(
+        dynamodb_client: &aws_sdk_dynamodb::Client,
+        settings: &Settings,
+        entry: &Self::Entry,
+    ) -> Result<(), Error> {
+        // Get table name.
+        let table_name = Self::table_name(settings);
+        // Convert Entry into the type needed for querying.
+        let entry_item: Item = serde_dynamo::to_item(entry)?;
+        // Add to the database.
+        dynamodb_client
+            .put_item()
+            .table_name(table_name)
+            .set_item(Some(entry_item.into()))
+            .send()
+            .await?;
+        // Return.
+        Ok(())
+    }
+
+    /// Get all entries from a dynamodb table.
+    async fn get_all_entries(
```

Comment:
I was thinking about it, but the whole feature flag is just to prevent possibly doomsday actions from being taken - this one doesn't have any mutation side effects so I'm not so worried about it.

Reply:
Yeah, that makes sense, but the reason why I mentioned it was because I was confused why it was even here. I read it and was like "wait getting all entries in a table is a bad idea", and then I remembered that it might only be used for testing. So I would either add the configure thing or add a comment to that effect.